api/src/main/java/org/apache/iceberg/actions/RepairManifests.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.actions;
+
+import org.apache.iceberg.ManifestFile;
+
+/** An action that will repair manifests. Implementations should produce a new set of manifests. */
+public interface RepairManifests extends SnapshotUpdate<RepairManifests, RepairManifests.Result> {
+
+  /**
+   * Configuration method for repairing manifest entry statistics
+   *
+   * @return this for method chaining
+   */
+  RepairManifests repairEntryStats();
+
+  /**
+   * Configuration method for removing duplicate file entries and removing files which no longer
+   * exist in storage
+   *
+   * @return this for method chaining
+   */
+  RepairManifests repairFileEntries();
```

Comment:
From the description, the interface only handles the missing or duplicated data files and does not include the missing manifest files.
This bug (https://github.com/apache/iceberg/issues/12792) accidentally deleted the manifest file, resulting in a corrupted table. I think we also need a separate interface `repairManifestEntries()` to fix the corrupted table caused by manifest issues.